src/main/java/io/cdap/plugin/servicenow/source/ServiceNowSourceConfig.java

```diff
@@ -395,72 +219,37 @@ private void validateTableQueryMode(FailureCollector collector) {
     if (Util.isNullOrEmpty(tableName)) {
       collector.addFailure("Table name must be specified.", null)
         .withConfigProperty(ServiceNowConstants.PROPERTY_TABLE_NAME);
+    } else {
+      // Call API to fetch first record from the table
+      ServiceNowTableAPIRequestBuilder requestBuilder = new ServiceNowTableAPIRequestBuilder(
+        this.getRestApiEndpoint(), tableName)
+        .setExcludeReferenceLink(true)
+        .setDisplayValue(this.getValueType())
+        .setLimit(1);
+
+      RestAPIResponse apiResponse = null;
+      ServiceNowTableAPIClientImpl serviceNowTableAPIClient = new ServiceNowTableAPIClientImpl(this);
+      try {
+        String accessToken = serviceNowTableAPIClient.getAccessToken();
+        requestBuilder.setAuthHeader(accessToken);
+
+        // Get the response JSON and fetch the header X-Total-Count. Set the value to recordCount
+        requestBuilder.setResponseHeaders(HEADER_NAME_TOTAL_COUNT);
+
+        apiResponse = serviceNowTableAPIClient.executeGet(requestBuilder.build());
+        if (!apiResponse.isSuccess()) {
+          if (apiResponse.getHttpStatus() == HttpStatus.SC_BAD_REQUEST) {
+            collector.addFailure("Bad Request. Table: " + tableName + " is invalid.", "")
+              .withConfigProperty(ServiceNowConstants.PROPERTY_TABLE_NAMES);
+          }
+        } else if (serviceNowTableAPIClient.parseResponseToResultListOfMap(apiResponse.getResponseBody()).isEmpty()) {
+          collector.addFailure("Table: " + tableName + " is empty.", "")
+            .withConfigProperty(ServiceNowConstants.PROPERTY_TABLE_NAMES);
+        }
+      } catch (OAuthSystemException | OAuthProblemException e) {
+        LOG.error("Error in fetchTableRecords", e);
```

Comment:
add failure to collector

Reply:
Errors added to the failure collector 7661e08